server-spi-private/src/main/java/org/keycloak/utils/StreamsUtil.java

```diff
@@ -142,4 +142,36 @@ public int characteristics() {
             }
         }, false);
     }
+
+    /**
+     * A Java stream utility prevents eager evaluation of the stream.
+     *
+     * @param <T> The type of the stream
+     * @param originalStream The original stream
+     * @return The stream that is lazily evaluating
+     */
+    public static <T> Stream<T> lazyStream(Stream<T> originalStream) {
+        Spliterator<T> source = originalStream.spliterator();
+        return StreamSupport.stream(new Spliterator<>() {
+            @Override
+            public boolean tryAdvance(Consumer<? super T> action) {
+                return source.tryAdvance(action);
+            }
+
+            @Override
+            public Spliterator<T> trySplit() {
+                return null;
+            }
+
+            public long estimateSize() {
+                return source.estimateSize();
+            }
+
+            @Override
+            public int characteristics() {
+                return ORDERED;
```

Comment:
Probably safer to use the source characteristics 
```suggestion
                return source.characteristics();
```